dev/NavigationView/NavigationView.cpp

```diff
@@ -714,6 +714,7 @@ void NavigationView::RaiseItemInvokedForNavigationViewItem(const winrt::Navigati
         auto inspectingDataSource = static_cast<InspectingDataSource*>(winrt::get_self<ItemsSourceView>(itemsSourceView));
         auto itemIndex = parentIR.GetElementIndex(nvi);
         nextItem = inspectingDataSource->GetAt(itemIndex);
+        SelectedItem(nextItem);
```

Comment:
We shouldn't be changing selection state in this method. The problem here is that ItemInvoked is being raised prematurely (at the beginning of OnNavigationViewItemInvoked).

I think the best way forward for this would be to raise this event in `ChangeSelection` based on a flag set in iteminvoked (something like `m_shouldRaiseInvokeItemInSelectionChange`).

However we need to be careful to still raise this event in OnNavigationViewItemInvoked if the item does not select on invoke or the user is invoking an already selected item (therefore not triggering a selection change).

Take a look at the placement of the RaiseItemInvoked calls before the move to repeater and it should help with understanding what I mean: [https://github.com/microsoft/microsoft-ui-xaml/blob/1f48f11b3cd895b10206f9abc93defb8eeff26b5/dev/NavigationView/NavigationView.cpp#L1391](https://github.com/microsoft/microsoft-ui-xaml/blob/1f48f11b3cd895b10206f9abc93defb8eeff26b5/dev/NavigationView/NavigationView.cpp#L1391)

Reply:
First of all, turns out I made a mistake during testing, the order inside OnNavigationViewItemInvoked was the cause of the issue.

Personally I am not a fan of raising ItemInvoked inside the ChangeSelection function since they are not really related and that would definitely add complexity to the already complex NavigationView.

Would changing the selection before we raise the event be a suitable solution? After all from the outside it behaves something like that:
1. User clicks on item
2. Item gets invoked
3. Item gets selected
4. ItemInvoked gets raised

I've updated the code now to change selection in a more appropriate place, looking forward to your feedback.

Reply:
The problem we have is that we want the ItemInvoked event to be raised before the SelectionChanged event. So with your new proposed change, the SelectionChanged event gets raised first.

I agree with you that it does not feel great to raise item invoke in `ChangeSelection`, however as of now, I don't see a better way of circumventing this problem.

Reply:
Yes that's a good point. I've moved it into ChangeSelection now with a separate flag.